src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -289,37 +299,76 @@ def process_ag_url_path_map_rule_create_namespace(namespace): # pylint: disable=
 
 def process_ag_create_namespace(namespace):
 
-    if namespace.public_ip:
+    # process folded parameters
+    if namespace.subnet or namespace.virtual_network_name:
+        get_subnet_validator(has_type_field=True, allow_new=True)(namespace)
+
+    if namespace.public_ip_address:
+        get_public_ip_validator(
+            has_type_field=True, allow_none=True, allow_new=True, default_none=True)(namespace)
         namespace.frontend_type = 'publicIp'
     else:
         namespace.frontend_type = 'privateIp'
         namespace.private_ip_address_allocation = 'static' if namespace.private_ip_address \
             else 'dynamic'
 
-    if not namespace.public_ip_type:
-        namespace.public_ip_type = 'none'
+    validate_cert(namespace)
 
 def process_auth_create_namespace(namespace):
     from azure.mgmt.network.models import ExpressRouteCircuitAuthorization
     namespace.authorization_parameters = ExpressRouteCircuitAuthorization()
 
 def process_lb_create_namespace(namespace):
-    if namespace.public_ip_dns_name:
-        namespace.dns_name_type = 'new'
 
     if namespace.subnet and namespace.public_ip_address:
-        raise argparse.ArgumentError(
-            None, 'Must specify a subnet OR a public IP address, not both.')
+        raise ValueError(
+            'incorrect usage: --subnet NAME_OR_ID [--vnet-name NAME] | --public-ip NAME_OR_ID')
```

Comment:
Should it be `'incorrect usage: --subnet NAME --vnet-name NAME | --subnet ID | --public-ip NAME_OR_ID`?